src/main/java/org/jabref/gui/desktop/JabRefDesktop.java

```diff
@@ -102,29 +106,32 @@ public static void openExternalViewer(BibDatabaseContext databaseContext,
                 if (eprintTypeOpt.isEmpty() && archivePrefixOpt.isEmpty()) {
                     dialogService.showErrorDialogAndWait(Localization.lang("Unable to open linked eprint. Please set the eprinttype field"));
                 } else {
-                    dialogService.showErrorDialogAndWait(Localization.lang("Unable to open linked eprint. Please verify that the eprint field has a valid '%0' id", eprintTypeOpt.get()));
+                    dialogService.showErrorDialogAndWait(Localization.lang("Unable to open linked eprint. Please verify that the eprint field has a valid '%0' id", link));
                 }
             }
             // should be opened in browser
-            field = StandardField.URL;
+            field = URL;
         }
 
-        if (StandardField.URL == field) {
-            openBrowser(link, preferencesService.getFilePreferences());
-        } else if (StandardField.PS == field) {
-            try {
-                NATIVE_DESKTOP.openFile(link, StandardField.PS.getName(), preferencesService.getFilePreferences());
-            } catch (IOException e) {
-                LOGGER.error("An error occurred on the command: " + link, e);
+        switch (field) {
+            case URL ->
+                    openBrowser(link, preferencesService.getFilePreferences());
+            case PS -> {
+                try {
+                    NATIVE_DESKTOP.openFile(link, PS.getName(), preferencesService.getFilePreferences());
+                } catch (IOException e) {
+                    LOGGER.error("An error occurred on the command: " + link, e);
+                }
             }
-        } else if (StandardField.PDF == field) {
-            try {
-                NATIVE_DESKTOP.openFile(link, StandardField.PDF.getName(), preferencesService.getFilePreferences());
-            } catch (IOException e) {
-                LOGGER.error("An error occurred on the command: " + link, e);
+            case PDF -> {
+                try {
+                    NATIVE_DESKTOP.openFile(link, PDF.getName(), preferencesService.getFilePreferences());
+                } catch (IOException e) {
+                    LOGGER.error("An error occurred on the command: " + link, e);
+                }
             }
-        } else {
-            LOGGER.info("Message: currently only PDF, PS and HTML files can be opened by double clicking");
+            case null, default ->
```

Comment:
Isn't null also matched by default?

Reply:
nope, null is an extra case https://blogs.oracle.com/javamagazine/post/java-switch-expression-arrow-case-null